From 8ae958d92cf46a5a7446e3993adca38a7b78d3ab Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm.code@gmail.com>
Date: Sat, 17 Feb 2018 00:58:21 +0100
Subject: [PATCH 1/2] toybox: Use toybox for dd, getprop and grep in recovery

 * dd and getprop are currently provided by toolbox in the main system.

 * grep is currently provided by BSD grep in the main system.

The toybox versions of these tools work well enough and we don't want
to give up the space needed to us the main system tools.

Change-Id: Ia7295ed991a8452c7b50d6d783a6a034f8e71365
---
 Android.mk | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/Android.mk b/Android.mk
index 3e00e4c0..8aa96131 100644
--- a/Android.mk
+++ b/Android.mk
@@ -397,6 +397,12 @@ ALL_TOOLS := \
     yes \
     zcat \
 
+ALL_RECOVERY_TOOLS := \
+    $(ALL_TOOLS) \
+    dd \
+    getprop \
+    grep
+
 ############################################
 # toybox for /system
 ############################################
@@ -440,5 +446,5 @@ LOCAL_STATIC_LIBRARIES := $(toybox_libraries)
 LOCAL_CXX_STL := libc++_static
 LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
 LOCAL_FORCE_STATIC_EXECUTABLE := true
-LOCAL_POST_INSTALL_CMD := $(hide) $(foreach t,$(ALL_TOOLS),ln -sf ${LOCAL_MODULE} $(LOCAL_MODULE_PATH)/$(t);)
+LOCAL_POST_INSTALL_CMD := $(hide) $(foreach t,$(ALL_RECOVERY_TOOLS),ln -sf ${LOCAL_MODULE} $(LOCAL_MODULE_PATH)/$(t);)
 include $(BUILD_EXECUTABLE)
-- 
2.17.1

